### DISCUSSION TITLE
Add peak memory consumption to results

### DISCUSSION THREAD

**Discussion by zigzag312** [jinyus/related_post_gen#341] [Ideas]
Memory consumption is important factor, but often missing in benchmarks.

**Comment by zupat** [comment]
This could be done by ditching hyperfine and just use the time command in a loop. I will look into this.

**Comment by zupat** [comment]
Done: https://github.com/jinyus/related_post_gen#memory-usage-results

**Reply by zigzag312** [reply]
Thank you! Very useful, because when you run out or RAM, performance tanks or you get a crash.

F# (AOT), while relatively fast, uses 1 GB for 60k posts. That's about 10x more than Rust. 

Scala native 4 GB for 60k posts. While Zig uses 30 MB. Ouch.

**Reply by zupat** [reply]
Yea, I was surprised by those 2 as they are my favorite Langs, though json might be influencing this number. I will try manually parsing to see if that could be the issue.

**Reply by zupat** [reply]
After doing some tests, I think I will have to remove the chart as the numbers are heavily skewed by json encoding/decoding and the solutions weren't optimized for that, so it's misleading.

**Reply by zigzag312** [reply]
Measuring memory consumption is hard. 

What if that chart is collapsed by default, and you add a warning that the numbers are heavily skewed by json encoding/decoding?
